Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -525,13 +525,33 @@ CGImageRef CGImageCreateCopyWithColorSpace(CGImageRef ref, CGColorSpaceRef color
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+ @Notes This function does not defer the composition of its mask.
+        This function supports non-grayscale alpha masks, unlike the reference platform.
  */
 CGImageRef CGImageCreateWithMask(CGImageRef image, CGImageRef mask) {
-    // TODO  #1425 - masks are applied during rendering via D2D.
     RETURN_NULL_IF(!image);
-    UNIMPLEMENTED();
-    return StubReturn();
+    RETURN_NULL_IF(!mask);
+    RETURN_NULL_IF(CGImageIsMask(image));
```

Comment:
See notes above; the reference platform requires a _mask_ (black/white image). We add support for an alpha clip (alpha image), but an alpha image _is not a mask_.